### PR TITLE
Add some sign that the link is to the API

### PR DESCRIPTION
**Pull request by RebbePod**
At first glance one would assume it should take you to the base text sensor esphome doc

## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by oxan** [comment]
That whole line is weird, I think moving it to the "See Also" section as an "API Reference" link (like so many other pages have) makes more sense.

**Comment by RebbePod** [comment]
Not sure if I did that right

**Review by oxan — APPROVED** [review]

File: components/text_sensor/custom.rst [hunk 1]
```diff
@@ -60,9 +60,8 @@ Configuration variables:
 
     - All options from :ref:`Text Sensor <config-text_sensor>`.
 
-See :apiclass:`TextSensor <text_sensor::TextSensor>`
-
 See Also
 --------
 
+- :apiclass:`TextSensor <text_sensor::TextSensor>`
```
Comment (oxan) [inline components/text_sensor/custom.rst]:
```suggestion
- :apiclass:`API Reference <text_sensor::TextSensor>`
```

**Comment by oxan** [comment]
Thanks!

**Pull request merged**